dlib/smart_pointers.h

```diff
@@ -13,10 +13,21 @@
 #pragma GCC warning "smart_pointers.h is included which will fail to compile under C++17"
 #endif
 
-#include "smart_pointers/scoped_ptr.h"
+#include <memory>
+
 #include "smart_pointers/shared_ptr.h"
 #include "smart_pointers/weak_ptr.h"
 
+namespace dlib {
+  // Template alias for compatibility with clients using old dlib::scoped_ptr
```

Comment:
Can you put this code into scoped_ptr.h instead?   Then keep the socket headers including scoped_ptr.h.  The reason being so that old dlib code that uses the sockets API will still compile.  With these current changes it looks like people who don't explicitly include the smart pointers header won't get the alias and then their socket code would fail to compile.